src/components/Table/Table.tsx

```diff
@@ -428,17 +428,21 @@ export class Table<I extends TableDataItem = Record<string, string>> extends Rea
         );
     }
 
-    private updateColumnWidths() {
-        this.setState({
-            columnsWidth: this.state.columnHeaderRefs.map((ref) =>
-                ref.current === null ? undefined : ref.current.getBoundingClientRect().width,
-            ),
-        });
+    private updateColumnStyles() {
+        const columnsWidth = this.state.columnHeaderRefs.map((ref) =>
+            ref.current === null ? undefined : ref.current.getBoundingClientRect().width,
+        );
+
+        const columnsStyles = this.props.columns.map((_, index) =>
+            this.getColumnStyles(index, columnsWidth),
+        );
+
+        this.setState({columnsStyles});
```

Comment:
Better use function version of setState (`setState((prevState) => newState)`) if you need previous state to build a new state